src/util/regulation.rs

```diff
@@ -88,7 +88,7 @@ impl Regulation {
 
         for file in &res.files {
             let path = PathBuf::from_str(&file.name).unwrap();
```

Comment:
You can drop the Pathbuf and just split the `file_name` directly. I believe it's a String or &str.